SparkPost/2web2ui
# FAD-6308 De-couple date management, reduxify suppressions and message events pages

The DateFilter and DateForm components that used to be inside src/pages/reports/components are now in src/components/datePicker to be shared app-wide, and are called DatePicker and ManualEntryForm, respectively.

To Do:
* [x] Write new tests for uncovered/new code paths
* [x] Add prop types to DatePicker component for documentation

To test:
* Visit the Message Events page and make sure all combinations of searches work as expected
* Visit the Suppressions Page and make sure all combinations of searches work as expected
* Run through all of the other reports pages and make sure their date management is still working as expected
* Verify that dates picked in the regular reports are now de-coupled from both message events and suppressions (and that message events and suppressions are not coupled to each other, either).
